provider/mem/mem.go

```diff
@@ -193,6 +212,8 @@ func (a *Alerts) GetPending() provider.AlertIterator {
 
 // Get returns the alert for a given fingerprint.
 func (a *Alerts) Get(fp model.Fingerprint) (*types.Alert, error) {
+	a.mtx.Lock()
```

Comment:
Same comment as @emanlodovice here. Would `RWMutex` help? If the intent is to prevent `gc` from running, acquiring read lock here would work, right? That way, concurrent `Get` can run, right?